src/wp-includes/html-api/class-wp-html-tag-processor.php

```diff
@@ -2281,6 +2512,134 @@ public function is_tag_closer() {
 		);
 	}
 
+	/**
+	 * Indicates the kind of matched token, if any.
+	 *
+	 * This differs from `get_token_name()` in that it always
+	 * returns a static string indicating the type, whereas
+	 * `get_token_name()` may return values derived from the
+	 * token itself, such as a tag name or processing
+	 * instruction tag.
+	 *
+	 * Possible values:
+	 *  - `#tag` when matched on a tag.
+	 *  - `#text` when matched on a text node.
+	 *  - `#cdata-section` when matched on a CDATA node.
+	 *  - `#comment` when matched on a comment.
+	 *  - `#presumptuous-tag` when matched on an empty tag closer.
+	 *  - `#funky-comment` when matched on a funky comment.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @return string|null What kind of token is matched, or null.
+	 */
+	public function get_token_type() {
+		switch ( $this->parser_state ) {
+			case self::STATE_MATCHED_TAG:
+				return '#tag';
+
+			case self::STATE_DOCTYPE:
+				return '#doctype';
```

Comment:
Why do we return `#doctype` here? The `html` value we'd get from `get_token_name` is confusing but aligns with what the browser does.

Reply:
this is specifically to expose what kind of token it is. I don't like conflating it with the `HTML` tag name for an Element, even though one is lower-case and the other upper-case.

in my own explorations I found it helpful to have both functions: one to say the node name like the browser would, and one to say the node type (also like how the browser would). I've also been trying to balance the use of longer constants against cearly-searchable text values since this is a more consumer-oriented function.

```php
switch ( $processor->get_token_type() ) {
	case WP_HTML_Processor::NODE_TYPE_DOCUMENT_TYPE:
	case '#doctype':
		…
}
```

at this point I'm assuming people will use the string value even if the constant exists. also I started with `get_node_type()` and `get_node_name()` but then renamed to `_token_` because I wanted to support a slightly different set of kinds; I'm doubting this since discovering the challenge of partial documents with invalid comment syntaxes, but haven't completely abandoned the idea yet, particularly because of the support for presumptuous tags and funky comments, which aren't in the DOM API.

Reply:
Thanks, I think this makes sense 👍